clang/include/clang/Basic/Attr.td

```diff
@@ -1545,6 +1550,18 @@ def SYCLKernel : InheritableAttr {
   let Documentation = [SYCLKernelDocs];
 }
 
+def GlobalStorageNonLocalVar : SubsetSubject<Var,
+                                             [{S->hasGlobalStorage() &&
+                                               !S->isLocalVarDeclOrParm()}],
+                                             "global variables">;
+
+def SYCLExternal : InheritableAttr {
+  let Spellings = [GNU<"sycl_external">];
```

Comment:
Since this attribute is a Clang extension, I think we should limit it to the Clang spelling for now as is done for the `sycl_kernel_entry_point` attribute.
```suggestion
  let Spellings = [Clang<"sycl_external">];
```